### ISSUE TITLE
Race condition in butlerCaller

### ISSUE THREAD

**Issue by fasterthanlime** [itchio/itch#2008]
Here's a fun problem that can happen (and just did in integration tests in front of me - at least that's what I think happened).

  * butlerCaller is mounted
  * launches query
  * gets local results, is marked stale
  * launches query with fresh
  * meanwhile, a button is clicked that changes the params
  * another query is launched with the new params
  * the query with the new params returns
  * the query with the old params + fresh returns
  * the old data is displayed

tl;dr we should ignore old results.
